app/controllers/sessions_controller.rb

```diff
@@ -0,0 +1,28 @@
+class SessionsController < ApplicationController
+    def login_receive
+        @user = User.new
+    end
+
+    def create
+        @user = User.new(:email => params[:email], :password=> params[:password])
+
+        if @user.valid?
+            user = User.where(email: @user.email).first
+
+            if user.nil?
+                @user.errors.add(:email, message: "This email is not registered")
+                render :login_receive, status: :unprocessable_entity
+            else
+                if user.password == @user.password
+                    session[:user_email] = @user.email
+                    redirect_to dashboard_index_path
+                else
+                    @user.errors.add(:email, message: "Wrong password")
+                    render :login_receive, status: :unprocessable_entity
+                end
+            end
+        else
+            render :login_receive, status: :unprocessable_entity
+        end
+    end
```

Comment:
I have a refactor suggestion:
```rb
    def create
      @user = User.new(:email => params[:email], :password=> params[:password])

      real_user = [@user].filter(|user| user.valid?)
        .map(|user| User.where(email: user.email).first)
        .filter(|user| user.nil?)
        .filter(|user| user.password == @user.password)
      
      if real_user.empty?
        @user.errors.add(:email, message: "E-mail or password could not be found")
        render :login_receive, status: :unprocessable_entity
      else
        session[:user_email] = @user.email
        redirect_to dashboard_index_path
      end
    end
```
It would change the error message the user would see, and I haven´t tested it, but it seems cleaner

Reply:
Can you take a new look on the recent changes and see if you still think that it's better?

Reply:
I don't think so.
![image](https://user-images.githubusercontent.com/61894952/200859294-79d1c9c9-1634-49f4-ba1e-b933241c262a.png)
(taken from [here](https://codeclimate.com/github/Fermented-Software/Kimchi/pull/31#))